modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb

```diff
@@ -111,6 +111,23 @@
         expect(response.body).to match_json_schema('VAOS_v2_appointments')
         expect(response.parsed_body['location']).to be_nil
       end
+
+      it 'does not attempt to fetch facility more than once' do
+        expect_any_instance_of(Mobile::AppointmentsHelper).to receive(:get_facility).with('983').once
+
+        VCR.use_cassette('mobile/appointments/VAOS_v2/get_clinic_200', match_requests_on: %i[method uri]) do
+          VCR.use_cassette('mobile/appointments/VAOS_v2/get_facility_500', match_requests_on: %i[method uri],
+                                                                           allow_playback_repeats: true) do
+            VCR.use_cassette('mobile/appointments/VAOS_v2/get_appointments_bad_facility_200',
+                             match_requests_on: %i[method uri]) do
+              get '/mobile/v0/appointments', headers: iam_headers, params:
+            end
+          end
+        end
+
+        expect(response.body).to match_json_schema('VAOS_v2_appointments')
+        expect(response.parsed_body['location']).to be_nil
```

Comment:
Why is this nil? I can see a few possible reasons. 
1) That's the value in the vcr. I haven't checked but I doubt it's empty.
2) Expect _any_instance_of sets a stub, which will generally prevent the method from being called. You may need to add `and_call_original` to the end of the stub in order to get the value.
2 It doesn't appear to be looking at a specific appointment. Is it just looking for location in the body rather than in a specific appointment?

Reply:
Oh right, I can add in `and_call_original`. I just copied the expectations from other tests that were testing a failed fetch to clinic and facility as this is the exact same tests, just now making sure it doesn't double perform the fetch. If you think this is wrong though, I can change those tests as well. The reason I point this out is I didn't make the decision on how to test this, I'm just using existing choices

Reply:
Hmm adding `and_call_original` actually is messing up the tests. It results in `response` not being set in the spec. Are you sure I need to call `and_call_original`?

Reply:
I'm not sure why response wouldn't be set. If anything, I'd expect a bad response, like a 500. 

Both of these tests are using 500 error vcrs. If you change those to successful cassettes, maybe it fixes them. 

For the purposes of these tests, I'm ok with not using `and_call_original` but in that case I don't think we should test the response body because it's just not relevant to the test and its presence makes the purpose of the tests less clear IMO.

Reply:
Good point, the intent of the test is to check the function calls, not the actual responses. Updated